tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -55,6 +55,232 @@ def _emit_event(self, resource, curr_value, msg, instance):
+                else:
+                    # this is a disperse volume, with all bricks online
+                    # assumption : all bricks are the same size
+                    disperse_yield = \
+                        float(
+                            int(volume.disperse_count) - int(volume.redundancy_count)
+                        ) / int(volume.disperse_count)
+                    vol_usable_capacity = raw_capacity * disperse_yield
+                    vol_used_capacity = raw_used * disperse_yield
+            else:
+                if int(volume.replica_count) > 1 or int(volume.disperse_count) > 0:
+                    for subvol in subvols:
+                        for brick in subvol:
+                            if brick.status == "Started":
+                                vol_usable_capacity += brick.utilization['total']
+                                vol_used_capacity += brick.utilization['used']
+                            # For replicate volume use only one replica
+                            if int(volume.replica_count) > 1:
+                                break
+                else:
+                    vol_usable_capacity = raw_capacity
+                    vol_used_capacity = raw_used
+            if vol_usable_capacity > 0:
+                vol_pcnt_used = (vol_used_capacity / float(vol_usable_capacity)) * 100
+            volume.usable_capacity = vol_usable_capacity
+            volume.used_capacity = vol_used_capacity
+            volume.pcnt_used = str(vol_pcnt_used)
+            volume.save()
+            cluster_usable_capacity += volume.usable_capacity
+        cluster_pcnt_used = 0
+        if cluster_usable_capacity > 0:
+            cluster_pcnt_used = (cluster_used_capacity / float(cluster_usable_capacity)) * 100
+
+        NS.gluster.objects.Utilization(
+            raw_capacity=cluster_raw_capacity,
+            used_capacity=cluster_used_capacity,
+            usable_capacity=cluster_usable_capacity,
+            pcnt_used=str(cluster_pcnt_used)
+        ).save()
+
+    def _sync_cluster_status(self, volumes):
+        status = 'healthy'
+
+        # Calculate status based on volumes status
+        if len(volumes) > 0:
+            volume_states = self._derive_volume_states(volumes)
+            for vol_id, state in volume_states.iteritems():
+                if 'down' in state or 'partial' in state:
+                    status = 'unhealthy'
+
+        # Change status basd on node status
+        cmd = cmd_utils.Command(
+            'gluster pool list', True
+        )
+        out, err, rc = cmd.run()
+        if not err:
+            out_lines = out.split('\n')
+            connected = True
+            for index in range(1, len(out_lines) - 1):
+                node_status_det = out_lines[index].split('\t')
+                if len(node_status_det) > 2:
+                    if node_status_det[2].strip() != 'Connected':
+                        connected = connected and False
```

Comment:
Need to do logical and with previous value. Say there are 4 nodes in the cluster and 3rd one is not connected so the final value should be like 
`connected = True and True and False and True` this effectively is False